odra-casper/casper-client/src/casper_client.rs

```diff
@@ -0,0 +1,648 @@
+use std::{fs, path::PathBuf, str::from_utf8_unchecked, time::Duration};
+
+use casper_execution_engine::core::engine_state::ExecutableDeployItem;
+use casper_hashing::Digest;
+use itertools::Itertools;
+use jsonrpc_lite::JsonRpc;
+use serde::de::DeserializeOwned;
+use serde_json::{json, Value};
+
+use odra_core::casper_types::URef;
+use odra_core::{
+    casper_types::{
+        bytesrepr::{Bytes, FromBytes, ToBytes},
+        runtime_args, ContractHash, ContractPackageHash, ExecutionResult, Key as CasperKey,
+        PublicKey, RuntimeArgs, SecretKey, TimeDiff, Timestamp, U512
+    },
+    consts::*,
+    Address, CLTyped, CallDef, ExecutionError, OdraError
+};
+
+use crate::casper_node_port::query_balance::{
+    PurseIdentifier, QueryBalanceParams, QueryBalanceResult, QUERY_BALANCE_METHOD
+};
+use crate::casper_node_port::rpcs::StoredValue::*;
+use crate::casper_node_port::{
+    rpcs::{
+        DictionaryIdentifier, GetDeployParams, GetDeployResult, GetDictionaryItemParams,
+        GetDictionaryItemResult, GetStateRootHashResult, GlobalStateIdentifier, PutDeployResult,
+        QueryGlobalStateParams, QueryGlobalStateResult
+    },
+    Deploy, DeployHash
+};
+use crate::{casper_node_port, log};
+
+pub const ENV_SECRET_KEY: &str = "ODRA_CASPER_LIVENET_SECRET_KEY_PATH";
+pub const ENV_NODE_ADDRESS: &str = "ODRA_CASPER_LIVENET_NODE_ADDRESS";
+pub const ENV_CHAIN_NAME: &str = "ODRA_CASPER_LIVENET_CHAIN_NAME";
+pub const ENV_ACCOUNT_PREFIX: &str = "ODRA_CASPER_LIVENET_KEY_";
+
+fn get_env_variable(name: &str) -> String {
+    std::env::var(name).unwrap_or_else(|err| {
+        log::error(format!(
+            "{} must be set. Have you setup your .env file?",
+            name
+        ));
+        panic!("{}", err)
+    })
+}
+
+/// Client for interacting with Casper node.
+pub struct CasperClient {
+    node_address: String,
+    chain_name: String,
+    active_account: usize,
+    secret_keys: Vec<SecretKey>,
+    gas: U512
+}
+
+impl CasperClient {
+    /// Creates new CasperClient.
+    pub fn new() -> Self {
+        // Check for additional .env file
+        let additional_env_file = std::env::var("ODRA_CASPER_LIVENET_ENV");
+
+        if let Ok(additional_env_file) = additional_env_file {
+            let filename = PathBuf::from(additional_env_file).with_extension("env");
+            dotenv::from_filename(filename).ok();
+        }
+
+        // Load .env
+        dotenv::dotenv().ok();
+
+        let mut secret_keys = vec![];
```

Comment:
Extract loading secret keys to separate method and describe it with example.

Reply:
Fixed